tests/CppUTestExt/IEEE754PluginTest.cpp

```diff
@@ -27,59 +27,67 @@
 
 /* The IEEE754ExceptionFlags plugin by definition requires C++11 */
 
-#ifdef CPPUTEST_COMPILER_FULLY_SUPPORTS_CXX11
-
 #include "CppUTest/CommandLineTestRunner.h"
 #include "CppUTest/TestHarness.h"
 #include "CppUTest/TestRegistry.h"
 #include "CppUTest/TestTestingFixture.h"
 #include "CppUTestExt/IEEE754ExceptionsPlugin.h"
-#include <cfenv>
 
 extern "C" { 
     #include "IEEE754PluginTest_c.h"
 }
 
+#if CPPUTEST_USE_STD_C_LIB
```

Comment:
You have a point, of course. Wrapping all the tests in one #if does make things simpler.

However, doing so makes it increasingly difficult to know at a glance whether certain tests were run or not. Always using IGNORE_TEST has two advantages:

1. The total number of tests for different configurations should always be the same. This can be verified at a glance.

2. The presence / absence of certain ignored tests is also easy to verify.

It is possible (and actually happened to me just now) that a #define that you thought to be 1 actually was 0 (because of wrong placement, before the CppUTest headers). So it can happen that the functionality is included, but the tests are not run. Without the ignored tests, most likely, you will never know. Fortunately, I had the opposite problem and got a linker error ;-).